src/Model/HasAttributesTrait.php

```diff
@@ -123,6 +134,12 @@ public function &getDocumentAttribute(string $key)
             return $this->mutableCache[$key];
         }
 
+        if ($caster = BaseDateTimeCast::resolve($this->casts[$key] ?? null)) {
+            $value = $caster->get($this->attributes[$key]);
+
+            return $value;
+        }
```

Comment:
Pensando bem, se existe um caster configurado e ele não é encontrado, o comportamento mais correto é o código dar erro.

```suggestion
        if ($casterName = $this->casts[$key] ?? null) {

            // dispara exceção em caso de caster não encontrado
            // implementar igual para os outros pontos que resolvem o caster.
            $caster = BaseDateTimeCast::resolve($casterName);

            $value = $caster->get($this->attributes[$key]);

            return $value;
        }
```